distribution/lib/Standard/Base/0.0.0-dev/src/Error.enso

```diff
@@ -161,3 +220,15 @@ type Error
     if_not_error : Any -> Any
     if_not_error self ~other =
         const self other
+
+## PRIVATE
+   Unwrap a wrapped error 0 or more times until the specified type is found.
+   Returns None if it is not found.
+look_for_wrapped_error : Any -> Any -> Maybe Any
+look_for_wrapped_error error_type=Any error_value =
+    if error_value.is_a error_type then Maybe.Some error_value else
+        try_unwrap =
+            inner_error = Wrapped_Error.from error_value . inner_error
+            if inner_error.is_a error_type then Maybe.Some inner_error else
+                look_for_wrapped_error error_type inner_error
```

Comment:
```suggestion
            @Tail_Call look_for_wrapped_error error_type inner_error
```

1. I think we should add Tail_Call because we can do so and so that will work better if there is a big chain of wrapped errors (although I assume that should be rare).
2. the `if` seems unnecessary if _the exact same condition_ is checked as the first line of the recursive `look_for_wrapped_error` call

Reply:
Yes, I was going to remove the redundant check, but I wanted to benchmark this first -- only because I'm modifying something so central, i.e. Error.catch. But perhaps it's overkill; I don't expect there's any code that uses catch heavily in a tight loop, but I wasn't sure.